tests/Common/TokensTest.php

```diff
@@ -866,4 +871,84 @@ public function testTokenWithoutTokensManagePermissionCanListAndViewOnlySelf()
             $this->assertEquals(403, $e->getCode());
         }
     }
+
+    public function testTokenWithoutCanPurgeTrashPermission()
+    {
+        $this->clearComponents();
+
+        // create token without purge trash permission
+        $options = (new TokenCreateOptions())
+            ->setCanManageBuckets(true)
+            ->setDescription('Token without canPurgeTrash permission')
+        ;
+
+        $tokenId = $this->_client->createToken($options);
+
+        $token = $this->_client->getToken($tokenId);
+        $this->assertFalse($token['canPurgeTrash']);
+
+        $client = new \Keboola\StorageApi\Client([
+            'token' => $token['token'],
+            'url' => STORAGE_API_URL,
+        ]);
+
+
+        $components = new Components($client);
+
+        $listOptions = (new ListComponentConfigurationsOptions())
+            ->setComponentId('provisioning')
+        ;
+
+        $configurations = $components->listComponentConfigurations($listOptions);
+        $this->assertCount(0, $configurations);
+
+        // create test configuration
+        $components->addConfiguration((new Configuration())
+            ->setComponentId('provisioning')
+            ->setConfigurationId('for-delete')
+            ->setConfiguration(['foo' => 'bar'])
+            ->setName('Main2'));
+
+        $configurations = $components->listComponentConfigurations($listOptions);
+        $this->assertCount(1, $configurations);
+
+        // move configuration to trash
+        $components->deleteConfiguration('provisioning', 'for-delete');
+
+        $configurations = $components->listComponentConfigurations($listOptions);
+        $this->assertCount(0, $configurations);
+
+        $listDeletedOptions = (new ListComponentConfigurationsOptions())
+            ->setComponentId('provisioning')
+            ->setIsDeleted(true)
+        ;
+
+        $configurations = $components->listComponentConfigurations($listDeletedOptions);
+        $this->assertCount(1, $configurations);
+
+        try {
+            $components->deleteConfiguration('provisioning', 'for-delete');
+            $this->fail('Token without permission should not delete configuration from trash');
+        } catch (ClientException $e) {
+            $this->assertEquals(403, $e->getCode());
```

Comment:
Vždycky matchovat i message, pokud je to obecné - 403 může být (byť tady není) to, že máš špatně test a třeba nemáš jen práva na zápis v projektu nebo tak něco.

Reply:
Taky se z toho pozná, když je špatně naformátovaná message - třeba chybí placeholder atp.

Reply:
Tady mi to prijde zbytecne? Ostatne dela se to tak v dalsich token testech.
Vsechno vraci obecnou message `You don't have access to the resource.`

Kdybych tu mel message `Nemas pravo smazat konfiguraci` tak by mi to smysl davalo.

To ze mam spatne test mi prijde je dost vylouceny naslednym prirazenim opravneni pro mazani kose a vyzkouseni ze mi potom mazani funguje?

Reply:
Aha, já jsem četl nejdřív testy, takže jsem čekal, že by tam mohla být nějaká hezčí message - aby se poznalo, že je to chyba, že nemáš purge nebo že nemáš přístup k projektu. 

Přijde mi, že to není ideální, že vlastně nevíš, co je špatně.

Reply:
https://keboola.slack.com/archives/C02C3GZUS/p1558088801000700